### ISSUE TITLE
usr/bin/fusermount-glusterfs: mount failed: Invalid argument on CoreOS 1465.7.0  + v1.7.3+coreos.0

### ISSUE THREAD

**Issue by ronakpandya7** [gluster/gluster-kubernetes#362]
Hello

we are facing problem creating pod on heketi glusterfs volume. We have created using `./gk-deploy -g` 
we have successfully created storageclass and pvc also volume also create in glusterfs.
But problem we face when we mount our data with pvc. below i have pasted error.
we used deployment kind to configure our pod. 

YAML 
```
apiVersion: v1
kind: Service
metadata:
  name: postgres
  namespace: ns-mgmt
  labels:
    app: postgres
spec:
  selector:
    app: postgres
  ports:
    - port: 5432
---
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: postgres
  namespace: ns-mgmt
  labels:
    app: postgres
spec:
  replicas: 1
  template:
    metadata:
      labels:
        app: postgres
    spec:
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: kubernetes.io/hostname
                operator: In
                values: ["worker07"]
      containers:
      - name: postgres
        image: postgres:9.6.4
        ports:
        - containerPort: 5432
        readinessProbe:
          tcpSocket:
            port: 5432
        volumeMounts:
        - mountPath: /var/lib/postgresql/data
          name: postgresmnt
        env:
        - name: POSTGRES_PASSWORD
          value: postgres
        - name: PGDATA
          value: /var/lib/postgresql/data/pgdata      
      volumes:
      - name: postgresmnt
        persistentVolumeClaim:
          claimName: sentry-pvc
```
endpoints:
```
default       heketi                         192.168.21.248:8080                                               
default       heketi-storage-endpoints       192.168.51.82:1,192.168.51.83:1,192.168.51.85:1                     
ns-mgmt       glusterfs-dynamic-sentry-pvc   192.168.51.82:1,192.168.51.83:1,192.168.51.85:1
```
PVC:
```
NAMESPACE   NAME         STATUS    VOLUME                                     CAPACITY   ACCESSMODES   STORAGECLASS   AGE
ns-mgmt     sentry-pvc   Bound     pvc-0775f0d6-9fa3-11e7-a627-00155d653701   10Gi       RWO           sentry-sc      59m
```

Error: 
```
Mounting arguments: 192.168.51.85:vol_089d183280f5989f0eee9aeab5132754 /var/lib/kubelet/pods/18623df0-9fa3-11e7-a627-00155d653701/volumes/kubernetes.io~glusterfs/pvc-0775f0d6-9fa3-11e7-a627-00155d6
53701 glusterfs [auto_unmount log-level=ERROR log-file=/var/lib/kubelet/plugins/kubernetes.io/glusterfs/pvc-0775f0d6-9fa3-11e7-a627-00155d653701/postgres-0-glusterfs.log backup-volfile-servers=192.
168.51.82:192.168.51.83:192.168.51.85]
Output: /usr/bin/fusermount-glusterfs: mount failed: Invalid argument
Mount failed. Please check the log file for more details.

 the following error information was pulled from the glusterfs log to help diagnose this issue: 
[2017-09-22 14:35:15.098133] I [socket.c:3578:socket_init] 0-vol_089d183280f5989f0eee9aeab5132754-client-1: SSL support is NOT enabled
[2017-09-22 14:35:15.098151] I [socket.c:3593:socket_init] 0-vol_089d183280f5989f0eee9aeab5132754-client-1: using system polling thread

  36m   2m      13      kubelet, worker02               Warning FailedMount     Unable to mount volumes for pod "postgres-0_ns-mgmt(18623df0-9fa3-11e7-a627-00155d653701)": timeout expired waiting f
or volumes to attach/mount for pod "ns-mgmt"/"postgres-0". list of unattached/unmounted volumes=[postgresmnt]
  36m   17s     17      kubelet, worker02               Warning FailedSync      Error syncing pod
  36m   17s     22      kubelet, worker02               Warning FailedMount     (combined from similar events): Unable to mount volumes for pod "postgres-0_ns-mgmt(18623df0-9fa3-11e7-a627-00155d653
701)": timeout expired waiting for volumes to attach/mount for pod "ns-mgmt"/"postgres-0". list of unattached/unmounted volumes=[postgresmnt]
```
any help appreciated.  

Thanks

**Comment by jarrpa** [comment]
@ronakpandya7 Your comment was difficult to read, so I updated your comment to use proper formatting. :) Please use proper GitHub markdown in the future.

It looks like you might be running into this: https://github.com/gluster/gluster-kubernetes/issues/334

Currently there is no workaround for that. The effort to get a proper solution merged is ongoing.

**Comment by humblec** [comment]
@ronakpandya7 This is fixed in kubernetes `1.8`
